kube/src/client/mod.rs

```diff
@@ -93,6 +114,34 @@ impl Client {
         Ok(res)
     }
 
+    /// Make WebSocket connection.
+    #[cfg(feature = "ws")]
+    pub async fn connect(&self, request: http::Request<()>) -> Result<WebSocketStream<ConnectStream>> {
+        let (mut parts, _) = request.into_parts();
+        if let Some(auth_header) = self.config.get_auth_header().await? {
+            parts.headers.insert(http::header::AUTHORIZATION, auth_header);
+        }
+        // Use the binary subprotocol v4.
+        // v4 is the current and sends JSON `metav1.Status` to `error` channel.
+        parts.headers.insert(
+            "sec-websocket-protocol",
+            "v4.channel.k8s.io".parse().expect("valid header value"),
+        );
+        // Replace scheme to ws(s).
+        let pandq = parts.uri.path_and_query().expect("valid path+query from kube");
+        parts.uri = finalize_url(&self.cluster_url, &pandq)
+            .replacen("http", "ws", 1)
+            .parse()
+            .expect("valid URL");
+        let (stream, _) = connect_async_with_tls_connector(
```

Comment:
Interestingly, there's also a `connect_async_with_tls_connector_and_config` variant that lets us set things like max frame_size/message_size/send_queue values. The [defaults](https://docs.rs/tungstenite/0.11.1/tungstenite/protocol/struct.WebSocketConfig.html) look sensible, but there's probably some memory implications there. Can you think of of reasons for/against including in `Config`?

Reply:
I can't think of reasons against it.
We should make it configurable along with the sizes of internal buffers of `AttachedProcess` to support different use cases.

Reply:
I made `AttachedProcess`'s internal buffer size configurable per request, but I think we can wait to make `WebSocketConfig` configurable until there's a need.